li-utils/src/main/java/com/linkedin/common/urn/DataJobUrn.java

```diff
@@ -0,0 +1,47 @@
+package com.linkedin.common.urn;
+
+import com.linkedin.common.FabricType;
+import java.net.URISyntaxException;
+
+import static com.linkedin.common.urn.UrnUtils.toFabricType;
+
+public class DataJobUrn extends Urn {
+    public static final String ENTITY_TYPE = "dataJob";
+
+    private final String dataJobNameEntity;
+
+    private static final String CONTENT_FORMAT = "(%s,%s,%s)";
+
+    private final String dataJobOrchestrator;
+
+    private final FabricType originEntity;
+
+    public DataJobUrn(String orchestrator, String name, FabricType origin) {
+        super(ENTITY_TYPE, String.format(CONTENT_FORMAT, orchestrator, name, origin.name()));
+        this.dataJobOrchestrator = orchestrator;
+        this.dataJobNameEntity = name;
+        this.originEntity = origin;
+    }
+
+    public String getJobNameEntity() {
+        return dataJobNameEntity;
+    }
+
+    public String getDataJobOrchestrator() {
+        return dataJobOrchestrator;
+    }
+
+    public FabricType getOriginEntity() {
+        return originEntity;
+    }
+
+    public static DataJobUrn createFromString(String rawUrn) throws URISyntaxException {
+        String content = new Urn(rawUrn).getContent();
+        String[] parts = content.substring(1, content.length() - 1).split(",");
+        return new DataJobUrn(parts[0], parts[1], toFabricType(parts[2]));
+    }
+
+    public static DataJobUrn deserialize(String rawUrn) throws URISyntaxException {
+        return createFromString(rawUrn);
+    }
```

Comment:
You'll most likely need to add the coercer as well like this: https://github.com/linkedin/datahub/blob/f9324377424209d74ba756d85a2dbc0839ca306d/li-utils/src/main/java/com/linkedin/common/urn/DatasetUrn.java#L53

Reply:
Don't you have removed `coercer`? I looked at the code base, all `coercer` files have been removed.

Reply:
Oh. I see. It is not remove, but combined.

Reply:
Yes there was an issue loading the non-embedded coercer correctly so we combined them all now.